### PR TITLE
Support reading data from TiDB in Spark Structured Streaming 

### PR DESCRIPTION
**Pull request by guliangliangatpingcap**
### What problem does this PR solve? 
close https://github.com/pingcap/tispark/issues/1102

When selecting data from tidb using spark structured streaming, the data selecting from tidb will never be updated.

The reason is that the `CoprocessorRDD's timestamp` will not be updated.

### What is changed and how it works?
Update `CoprocessorRDD's timestamp` if in Streaming Context and apply on `Root LogicalPlan`.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
https://github.com/pingcap/tispark-test/blob/master/tispark-examples/src/main/scala/com/pingcap/tispark/examples/StructuredStreamingReadTiDB.scala

Related changes

 - Need to cherry-pick to the release branch 
 - Need to be included in the release note

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/TiStrategy.scala [hunk 1]
```diff
@@ -44,6 +44,20 @@ import org.joda.time.{DateTime, DateTimeZone}
...
 import scala.collection.mutable
 
+object TiStrategy {
+  private val assignedTSPlanCache = new mutable.WeakHashMap[LogicalPlan, Boolean]()
+
+  private def hasAssignedTS(plan: LogicalPlan): Boolean = {
+    assignedTSPlanCache.get(plan).isDefined
+  }
+
+  private def markAssignedTS(plan: LogicalPlan): Unit = {
```
Comment (zhexuany) [inline core/src/main/scala/org/apache/spark/sql/TiStrategy.scala]: markTSAssigned?

Comment (guliangliangatpingcap) [reply core/src/main/scala/org/apache/spark/sql/TiStrategy.scala]: nice

File: core/src/main/scala/org/apache/spark/sql/TiStrategy.scala [hunk 2]
```diff
@@ -115,7 +130,16 @@ case class TiStrategy(getOrCreateTiContext: SparkSession => TiContext)(sparkSess
 
   override def apply(plan: LogicalPlan): Seq[SparkPlan] = {
     val ts = tiContext.tiSession.getTimestamp
-    plan foreachUp applyStartTs(ts)
+
+    if (plan.isStreaming) {
```
Comment (zhexuany) [inline core/src/main/scala/org/apache/spark/sql/TiStrategy.scala]: I feel like we need some comment to explain why we do this.

Comment (guliangliangatpingcap) [reply core/src/main/scala/org/apache/spark/sql/TiStrategy.scala]: nice!

**Pull request merged**